src/test/shared/credentials/userCredentialsUtils.test.ts

```diff
@@ -151,6 +150,11 @@ describe('UserCredentialsUtils', () => {
             assert(profiles)
             assert(profiles.credentialsFile)
             assert(profiles.credentialsFile[profileName])
+
+            const { mode } = fs.statSync(credentialsFilename)
```

Comment:
`filesystem.ts` has `statAsync` that you can await instead of using the Sync version

Reply:
1. Node 10 :) We can now use: https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fspromises_stat_path_options
2. Since this is a test and our Mocha tests are run in series, I don't know that there is really advantage to using async.. but true async is a better default